File: lib/src/main/java/com/example/lib/Player.java
package com.example.lib;

public class Player {

    enum State{
        IDLE,PLAYING,PAUSED
    }

    private State state=State.IDLE;

    public void pause(){
       switch (state){
           case IDLE:
           case PAUSED:
               //do somethings
           case PLAYING:
               break;
       }
    }


}

File: lib/src/main/java/com/example/lib/Player.kt
package com.example.lib

class Player {
    internal enum class State {
        IDLE, PLAYING, PAUSED
    }

    private val state = State.IDLE
    fun pause() {
        when (state) {
            State.IDLE,
            State.PAUSED,

            State.PLAYING -> {
            }
        }
    }
}